codec-redis/src/main/java/io/netty/handler/codec/redis/RedisDecoder.java

```diff
@@ -0,0 +1,294 @@
+            return true;
+        case BULK_STRING:
+            if (length > RedisConstants.REDIS_MESSAGE_MAX_LENGTH) {
+                throw new RedisCodecException("length: " + length + " (expected: <= " +
+                                              RedisConstants.REDIS_MESSAGE_MAX_LENGTH + ")");
+            }
+            state = State.DECODE_BULK_STRING_HEADER;
+            expectedBulkLength = (int) length; // range(int) is already checked.
+            toReadBulkLength = expectedBulkLength;
+            out.add(new BulkStringHeaderRedisMessage(expectedBulkLength));
+            return true;
+        default:
+            throw new RedisCodecException("bad type: " + type);
+        }
+    }
+
+    private boolean decodeBulkString(ByteBuf in, List<Object> out) throws Exception {
+        switch (expectedBulkLength) {
+        case RedisConstants.NULL_RESPONSE_LENGTH:
+            // $-1\r\n
+            out.add(FullBulkStringRedisMessage.NULL_INSTANCE);
+            resetDecoder();
+            return true;
+        case 0:
+            // $0\r\n <here> \r\n
+            if (in.readableBytes() < RedisConstants.EOL_LENGTH) {
+                return false;
+            }
+            readEndOfLine(in);
+            out.add(FullBulkStringRedisMessage.EMPTY_INSTANCE);
+            resetDecoder();
+            return true;
+        default: // expectedBulkLength is always positive.
+            state = State.DECODE_BULK_STRING_CONTENT;
+            return decodeBulkStringContent(in, out);
+        }
+    }
+
+    // ${expectedBulkLength}\r\n <here> {data...}\r\n
+    private boolean decodeBulkStringContent(ByteBuf in, List<Object> out) throws Exception {
+        if (!in.isReadable()) {
+            return false;
+        }
+
+        // if this is last frame.
+        if (in.isReadable(toReadBulkLength + RedisConstants.EOL_LENGTH)) {
+            ByteBuf content = in.readSlice(toReadBulkLength).retain();
+            readEndOfLine(in);
+            out.add(new DefaultLastBulkStringRedisContent(content));
+            resetDecoder();
+            return true;
+        }
+
+        // chunked write.
+        int toRead = Math.min(toReadBulkLength, in.readableBytes());
+        toReadBulkLength -= toRead;
+        out.add(new DefaultBulkStringRedisContent(in.readSlice(toRead).retain()));
+        return true;
+    }
+
+    private static void readEndOfLine(final ByteBuf in) {
+        if (RedisConstants.EOL_SHORT == in.readShort()) {
+            return;
+        }
+        throw new RedisCodecException("expected: \\r\\n");
+    }
+
+    private RedisMessage newInlineRedisMessage(RedisMessageType messageType, ByteBuf content) {
+        switch (messageType) {
+        case SIMPLE_STRING: {
+            SimpleStringRedisMessage cached = messagePool.getSimpleString(content);
+            return cached != null ? cached : new SimpleStringRedisMessage(content.toString(CharsetUtil.UTF_8));
+        }
+        case ERROR: {
```

Comment:
False positive. I've just deactivated this rule so they does not appear from the next analysis.